backend/vxlan/vxlan_windows.go

```diff
@@ -142,12 +141,24 @@ func (be *VXLANBackend) RegisterNetwork(ctx context.Context, wg sync.WaitGroup,
 					}
 					remoteDrMac = policySettings.Address
 				}
+				if policy.Type == hcn.ProviderAddress {
+					policySettings := hcn.ProviderAddressEndpointPolicySetting{}
+					err = json.Unmarshal(policy.Settings, &policySettings)
+					if err != nil {
+						return nil, fmt.Errorf("Failed to unmarshal settings")
+					}
+					providerAddress = policySettings.ProviderAddress
+				}
+			}
+			if providerAddress != be.extIface.ExtAddr.String() {
+				log.Infof("Cannot use DR MAC %v since PA %v does not match %v", remoteDrMac, providerAddress, be.extIface.ExtAddr.String())
+				remoteDrMac = ""
 			}
 		}
 	}
```

Comment:
At the exit of the for loop, can you add a check for the availability of providerAddress & remoteDrMac. Fail, with a meaningful error if you do not have them, since you cannot proceed.

Reply:
Added below